### ISSUE TITLE
Support setting SimpleExoPlayerView background/shutter color

### ISSUE THREAD

**Issue by or-dvir** [google/ExoPlayer#3172]
**- Search existing issues, including issues that are closed.**
i only found this thread: https://github.com/google/ExoPlayer/issues/2821
but the question there is regarding surface view, and not exoPlayer components.
**- Consult our FAQs, supported devices and supported formats pages.
did not find my answer there

### Issue description
i would like to change the background of SimpleExoPlayerView.
right now, while the video is loading for the first time (meaning not buffering in the middle of streaming)
the background is black. i would like to change it to white.
using `android:background` in xml or `exoPlayerView.setBackground()` in code does not work.
changing the foreground does change the color, but it also blocks (or hides) all the other UI element, including the video.

### Reproduction steps
play any video that takes time to initially load

### Version of ExoPlayer being used
2.5.1

### Device(s) and version(s) of Android being used
LG G3 API 23.
LG K10 API 23.
Uniden prime C471 API 18.

in the layout file `exo_simple_player_view.xml`, which is found here: https://github.com/google/ExoPlayer/blob/release-v2/library/ui/src/main/res/layout/exo_simple_player_view.xml
there is a view with id "exo_shutter" which has a black background - im assuming this is the cause of my problem?
some might say to just do this (pseudo code):
```
int resId = getResources().getIdentifier("exo_shutter", "id", getPackageName());
findViewById(resId).setBackground(Color.white);
```
but it is not good practice to relay of inner id's like that... if they change in a future version i will get null pointer exception.
**unless,** can you guarantee that this will not be changed in the future and is safe to do?

**Comment by ojw28** [comment]
Yes, the shutter is the problem and your suggested approach of relying on exo_shutter will work for now. It is actually a documented id (see [here](https://google.github.io/ExoPlayer/doc/reference/com/google/android/exoplayer2/ui/SimpleExoPlayerView.html)), so it's not quite has hacky as you might think, although I agree it's definitely not ideal.

Question: Are you attempting to change the color only when the shutter is visible (e.g. when there's no video playing at all), or do you also want to change the color of any letter-boxing bars around the video during playback?

**Comment by or-dvir** [comment]
thanks! that link you provided pretty much answers my question. at least i think it does... before i try it i want to make sure i understood correctly.
right now i am overriding the controllers and i made a file called `exo_playback_control_view.xml`,
and that file is the **entire** layout that is being inflated as my video controllers.
if i understood correctly from the docs you sent (please correct me if im wrong) i need to create a file called `exo_simple_player_view.xml` and inside it create a single view with id `exo_shutter` and then simply change its' background to whatever color i want. please keep in mind that i **only** want to change this shutter view, and keep everything else as the default (`exo_content_frame`, `exo_subtitles`, `exo_artwork` etc). 
unlike the controllers layout file which is inflated in it's entirety, exoPlayer looks for specific id's inside `exo_simple_player_view.xml` and if they are not found then it goes to a default implementation?
is this correct?


as for your question, in my specific use case i am not using letter-boxing (that is until management will change their minds) so i was only referring to the first case "when the shutter is visible". however coloring the letter-boxing is a good idea in my opinion - the more control i have as a developer the better :)

**Comment by ojw28** [comment]
Overriding `exo_simple_player_view.xml` is similar to overriding `exo_playback_control_view.xml` (you need to provide the entire layout in both cases). Both are described in this [blog post](https://medium.com/google-exoplayer/customizing-exoplayers-ui-components-728cf55ee07a) (although note that some of the information may be a little out of date, since the library has evolved a little since then).

This is marked as an enhancement because we should just provide a way to programmatically change the shutter color, and we'll do that at some point. Until then, your original approach of changing it programmatically using the view ID is probably the simplest thing you can do.

**Comment by or-dvir** [comment]
thanks a lot for your help.
i did it and it worked

**Comment by ojw28** [comment]
Additional change: e92667ba5b58cbda9695c8168d3f8fa810da8288

**Comment by hare989** [comment]
Hello, I'm not sure if I should create new issue for this ( I've created issue #3741 ) but setting SimpleExoPlayerView.setShutterBackgroundColor(Color.TRANSPARENT) seems that doesn't work on couple Samsung devices. I found that on Samsung Galaxy S7 Edge (SM-G935F) Android 7, Samsung Galaxy J5 (Android 6) when preparing player (changing source) shutter is black although should be transparent and make last viewing frame vissible. It's working fine on other Android 6, 7, 8 devices/emulators. (Huawei P9 Lite Android 6, Xiaomi mi 4 Android 7.1) 

I tried also [customizing ui components](https://medium.com/google-exoplayer/customizing-exoplayers-ui-components-728cf55ee07a)

This happens with 2.6.0 ExoPlayer.
